monitoring/exporter/project_data.go

```diff
@@ -0,0 +1,159 @@
+package exporter
+
+import (
+	"fmt"
+	"time"
+
+	"go.opencensus.io/tag"
+	"google.golang.org/api/support/bundler"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// maximum number of time series that stackdriver accepts. Only test may change this value.
+var MaxTimeSeriesPerUpload = 200
+
+// projectData contain per-project data in exporter. It should be created by newProjectData()
+type projectData struct {
+	parent    *StatsExporter
+	projectID string
+	// We make bundler for each project because call to monitoring RPC can be grouped only in
+	// project level
+	bndler expBundler
+}
+
+// We wrap bundler and its maker for testing purpose.
+type expBundler interface {
```

Comment:
Again, I feel this is not the correct use of interface. You should return concrete types wherever possible, and interfaces should be defined by the users at the place of use.
https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a

Reply:
ok. Similar to monitroing.MetricClient, I mocked some function defined in the blunder package.